src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return width == null && length == null && color == null
+                ? rectangle.color == null : color.equals(rectangle.color);
```

Comment:
why are you checking only color field here?

Reply:
changed